infra/main.tf

```diff
@@ -0,0 +1,67 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.48.0"
+    }
+  }
+}
+
+provider "aws" {
+  region = "eu-central-1"
+  default_tags {
+    tags = {
+      Owner = "Jovan Drobnjak"
+    }
+  }
+  access_key = var.aws_access_key
+  secret_key = var.aws_secret_access_key
+  token      = var.aws_session_token
+}
+
+module "subnet_addrs" {
+  source          = "hashicorp/subnets/cidr"
+  base_cidr_block = "172.18.0.0/16"
+  networks = [
+    {
+      name     = "private-a"
+      new_bits = 2
+    },
+    {
+      name     = "private-b"
+      new_bits = 2
+    },
+    {
+      name     = "private-c"
+      new_bits = 2
+    },
+    {
+      name     = "public-a"
+      new_bits = 8
+    },
+    {
+      name     = "public-b"
+      new_bits = 8
+    },
+    {
+      name     = "public-c"
+      new_bits = 8
+    },
+  ]
+}
+
+module "vpc" {
+  source = "terraform-aws-modules/vpc/aws"
+
+  name = "vpc-jovand-02"
+  cidr = module.subnet_addrs.base_cidr_block
+  azs  = data.aws_availability_zones.available_zones.names
+
+  private_subnets        = [for key in local.private_subnet_keys : module.subnet_addrs.network_cidr_blocks[key]]
```

Comment:
To avoid having separate variable just for iteration you can do something like this:

  private_subnets = [
    lookup(module.subnet_addrs.network_cidr_blocks, "private-a", "what?"),
    lookup(module.subnet_addrs.network_cidr_blocks, "private-b", "what?"),
    lookup(module.subnet_addrs.network_cidr_blocks, "private-c", "what?"),
  ]
  
  https://developer.hashicorp.com/terraform/language/functions/lookup